src/main/java/com/blazemeter/taurus/junit/generator/Worker.java

```diff
@@ -0,0 +1,101 @@
+package com.blazemeter.taurus.junit.generator;
+
+import com.blazemeter.taurus.junit.CustomRunner;
+import com.blazemeter.taurus.junit.JUnitRunner;
+import com.blazemeter.taurus.junit.reporting.TaurusReporter;
+import com.blazemeter.taurus.junit4.JUnit4Runner;
+import com.blazemeter.taurus.junit5.JUnit5Runner;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static com.blazemeter.taurus.junit.CustomRunner.*;
+
+public class Worker extends Thread {
+    private static final Logger log = Logger.getLogger(CustomRunner.class.getName());
+
+    private final Properties props = new Properties();
+    private final List<Class> classes = new ArrayList<>();
+    private final TaurusReporter reporter;
+
+    private long iterations;
+    private long workingTime;
+    private long startDelay;
+
+    private JUnitRunner runner;
+    private Object request;
+
+    public Worker(List<Class> classes, Properties properties, TaurusReporter reporter, long startDelay) {
+        this.props.putAll(properties);
+        this.classes.addAll(classes);
+        this.reporter = reporter;
+
+        this.startDelay = startDelay * 1000;
+        float rampUp = Float.valueOf(props.getProperty(RAMP_UP, "0"));
+        float hold = Float.valueOf(props.getProperty(HOLD, "0"));
```

Comment:
Meaning of Hold-for is to form load pattern for whole test. So it has to be handled on Supervisor level. It should tell threads to finish once overall hold-for time is reached.